framework/sac_rlhf.py

```diff
@@ -489,7 +489,7 @@ def get_action(self, x):
         env=envs,
     ).to(device)
     reward_optimizer = optim.Adam(
-        reward_net.parameters(), lr=args.teacher_learning_rate
+        reward_net.parameters(), lr=args.teacher_learning_rate, weight_decay=1e-4
```

Comment:
This is the crutial part: `weight_decay=1e-4 ` adds a small L2 penalty on the model parameters. It directly penalizes large weights, encouraging the network to find simpler (smoother) solutions